### ISSUE TITLE
Fix dockerhub images (other than `latest`)

### ISSUE THREAD

**Issue by asmacdo** [dandi/dandi-hub#219]
When launching `latest-gpu`, we get the following traceback on the user pod.

```
Defaulted container "notebook" out of: notebook, nfs-fixer (init), block-cloud-metadata (init)
Entered start.sh with args: start-notebook.py
Running hooks in: /usr/local/bin/start-notebook.d as uid: 1000 gid: 100
Done running hooks in: /usr/local/bin/start-notebook.d
WARNING: container must be started as root to grant sudo permissions!
Running hooks in: /usr/local/bin/before-notebook.d as uid: 1000 gid: 100
Sourcing shell script: /usr/local/bin/before-notebook.d/10activate-conda-env.sh
Done running hooks in: /usr/local/bin/before-notebook.d
Executing the command: start-notebook.py
Traceback (most recent call last):
  File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/sessions/sessionmanager.py", line 14, in <module>
    import sqlite3
  File "/opt/conda/lib/python3.11/sqlite3/__init__.py", line 57, in <module>
    from sqlite3.dbapi2 import *
  File "/opt/conda/lib/python3.11/sqlite3/dbapi2.py", line 27, in <module>
    from _sqlite3 import *
ImportError: /opt/conda/lib/python3.11/lib-dynload/_sqlite3.cpython-311-x86_64-linux-gnu.so: undefined symbol: sqlite3_deserialize

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/jupyterhub-singleuser", line 6, in <module>
    from jupyterhub.singleuser import main
  File "/opt/conda/lib/python3.11/site-packages/jupyterhub/singleuser/__init__.py", line 65, in <module>
    from .extension import main
  File "/opt/conda/lib/python3.11/site-packages/jupyterhub/singleuser/extension.py", line 37, in <module>
    from jupyter_server.extension.application import ExtensionApp
  File "/opt/conda/lib/python3.11/site-packages/jupyter_server/extension/application.py", line 16, in <module>
    from jupyter_server.serverapp import ServerApp
  File "/opt/conda/lib/python3.11/site-packages/jupyter_server/serverapp.py", line 97, in <module>
    from jupyter_server.gateway.managers import (
  File "/opt/conda/lib/python3.11/site-packages/jupyter_server/gateway/managers.py", line 32, in <module>
    from ..services.sessions.sessionmanager import SessionManager
  File "/opt/conda/lib/python3.11/site-packages/jupyter_server/services/sessions/sessionmanager.py", line 17, in <module>
    from pysqlite2 import dbapi2 as sqlite3  # type:ignore[no-redef]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'pysqlite2'
```

I wasn't able to replicate this locally exactly, but I did see a similar issue.

### Confirming `latest`

`podman run -p 8000:8000 -d --name jupyterhub dandiarchive/dandihub:latest jupyterhub --no-ssl`

This launches the image locally, and I can access via `localhost:8000`. I wasn't able to login, but I think thats ok.

I stopped and removed 

### Testing `latest-gpu`

`podman run -p 8000:8000 -d --name jupyterhub dandiarchive/dandihub:latest-gpu jupyterhub --no-ssl`

The image started and crashed. 

`podman logs jupyterhub`

```
[I 2024-12-18 18:11:35.031 JupyterHub app:2859] Running JupyterHub version 4.0.2
[I 2024-12-18 18:11:35.031 JupyterHub app:2889] Using Authenticator: jupyterhub.auth.PAMAuthenticator-4.0.2
[I 2024-12-18 18:11:35.031 JupyterHub app:2889] Using Spawner: jupyterhub.spawner.LocalProcessSpawner-4.0.2
[I 2024-12-18 18:11:35.031 JupyterHub app:2889] Using Proxy: jupyterhub.proxy.ConfigurableHTTPProxy-4.0.2
[I 2024-12-18 18:11:35.053 JupyterHub app:1709] Writing cookie_secret to /home/jovyan/jupyterhub_cookie_secret
[E 2024-12-18 18:11:35.057 JupyterHub app:3382]
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.11/site-packages/jupyterhub/app.py", line 3379, in launch_instance_async
        await self.initialize(argv)
      File "/opt/conda/lib/python3.11/site-packages/jupyterhub/app.py", line 2901, in initialize
        self.init_db()
      File "/opt/conda/lib/python3.11/site-packages/jupyterhub/app.py", line 1838, in init_db
        self.session_factory = orm.new_session_factory(
                               ^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/jupyterhub/orm.py", line 1072, in new_session_factory
        engine = create_engine(url, **kwargs)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      File "<string>", line 2, in create_engine
      File "/opt/conda/lib/python3.11/site-packages/sqlalchemy/util/deprecations.py", line 281, in warned
        return fn(*args, **kwargs)  # type: ignore[no-any-return]
               ^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/sqlalchemy/engine/create.py", line 601, in create_engine
        dbapi = dbapi_meth(**dbapi_args)
                ^^^^^^^^^^^^^^^^^^^^^^^^
      File "/opt/conda/lib/python3.11/site-packages/sqlalchemy/dialects/sqlite/pysqlite.py", line 505, in import_dbapi
        from sqlite3 import dbapi2 as sqlite
      File "/opt/conda/lib/python3.11/sqlite3/__init__.py", line 57, in <module>
        from sqlite3.dbapi2 import *
      File "/opt/conda/lib/python3.11/sqlite3/dbapi2.py", line 27, in <module>
        from _sqlite3 import *
    ImportError: /opt/conda/lib/python3.11/lib-dynload/_sqlite3.cpython-311-x86_64-linux-gnu.so: undefined symbol: sqlite3_deserialize
 ```

**Comment by kabilar** [comment]
Thanks @asmacdo.  So do we need to add `pysqlite2` to the images?  Is it a dependency of JupyterHub that wasn't included in a requirements file upstream?

**Comment by satra** [comment]
you may want to check if you can update to a more recent jupyter image. this was an issue with sqlite earlier in the year.

**Comment by kabilar** [comment]
Thanks.  I am working on rebuilding and testing the images now.

**Comment by kabilar** [comment]
I created a new image with the base `quay.io/jupyter/datascience-notebook:python-3.11.10` and locally ran `/usr/local/bin/start.sh start-notebook.py` within the container, which was able to successfully start the Jupyter Server.  I will merge #216 in the morning with these changes and trigger a new release of the images.

**Comment by kabilar** [comment]
I was able to locally build and push the image to Docker Hub with some changes to the Dockerfile - see #216.  I can confirm that I can now can launch a user pod on DANDI Hub with a T4 GPU instance.  cc @tekrajchhetri 

Unfortunately, when I use GitHub Actions to build and push the image, I run into an error.  See [latest run](https://github.com/kabilar/dandi-hub/actions/runs/12437461580/job/34727366856).  Working on resolving.

**Comment by kabilar** [comment]
All the images except for the Matlab ones now are able to build with GitHub Actions.

**Comment by tekrajchhetri** [comment]
Thanks @kabilar for the update. I was able to successfully start the T4 GPU instance. I will update if there's any issue.

**Comment by kabilar** [comment]
Fixed in #216.
